engine/src/lib.rs

```diff
@@ -670,7 +672,7 @@ mod contract {
         io.assert_private_call().sdk_unwrap();
 
         let args: SetContractDataCallArgs = io.read_input_borsh().sdk_unwrap();
-        EthConnectorContract::set_contract_data(&mut io, args);
+        EthConnectorContract::set_contract_data(&mut io, args).sdk_expect("ERR_INVALID_DATA");
```

Comment:
As stated earlier, if this returns an error and not an option, we don't need to use `sdk_expect` here.